test/integration/testLargeResultSet.js

```diff
@@ -106,3 +107,43 @@ describe('Large result Set Tests', function ()
     });
   });
 });
+
+describe('SNOW-743920: Large result set with ~35 chunks', function () {
+  let connection;
+  const tableName = 'test_table';
+  const sourceRowCount = 251002;
+  const createTable = `create or replace table ${tableName} (data string)`;
+  const populateData = `insert into ${tableName} select randstr(20, random()) from table (generator(rowcount =>${sourceRowCount}))`;
+  const selectAllFromOrders = `select * from ${tableName}`;
+
+  before(async () => {
+    connection = testUtil.createConnection();
+    await testUtil.connectAsync(connection);
+    await testUtil.executeCmdAsync(connection, 'alter session set ROWS_PER_RESULTSET = 1000000');
+    await testUtil.executeCmdAsync(connection, 'alter session set USE_CACHED_RESULT = false;');
+    await testUtil.executeCmdAsync(connection, createTable);
+    await testUtil.executeCmdAsync(connection, populateData);
+    configureLogger('TRACE');
+  });
+
+  after(async () => {
+    configureLogger('ERROR');
+    await testUtil.dropTablesIgnoringErrorsAsync(connection, [tableName]);
```

Comment:
What if this line ends with error? Will the connection be closed?

Reply:
it shouldn't end with error as it ignores errors :)